lib/errors.ts

```diff
@@ -223,3 +233,14 @@ export function exitWithExpectedError(message: string | Error): never {
 	printErrorMessage(message);
 	process.exit(1);
 }
+
+// Support stubbing of module functions.
+export default ErrorsModule = {
```

Comment:
In the case of getSentry(), that is correct, but it depends on the function. Some are already public, but need to also be exported this way to support testing.  The advantage is that then these individual elements can be stubbed, allowing granular unit testing.  Otherwise there's no way to substitute stubs/mocks in the tests.

Reply:
> I think the CI says that the commit is "invalid" because it is missing a blank space after the colon in the line: `Resolves:#1807`

Agr that again.  Is this a balena CI thing?

Reply:
> Otherwise there's no way to substitute stubs/mocks in the tests

Perhaps using [rewire](https://www.npmjs.com/package/rewire)? It is already included as a `devDependency` in `package.json`.

> Is this a balena CI thing?

Probably [resin-commit-lint](https://github.com/balena-io-modules/resin-commit-lint), which is closely related to versionbot that updates the changelog file. I think.

Reply:
@pdcastro  Yeah rewire is worth looking into, I've used it before, but I think it doesn't work very well with ts/transpilers and `import`.  I think we were talking about only using js for tests - perhaps we now have another good reason to do that.

Ah right, I might try and see if it's possible to PR `resin-commit-lint` to make that syntax less picky.

Reply:
Our tests currently test the `build` folder where everything is javascript, right? The code to be "rewired" would be javascript in the `build` folder. The test code itself is mostly typescript but we could add javascript files, I don't have anything against it, especially if it saves time. Plain `require` (instead of `import`) alongside `ts-ignore` annotations can be added to typescript files too, in case the problem with `rewire` is the ts-node / tsc compiler type checking. (I'm saying this just in case it helps, I am not asking for any specific changes to be made.)